taskvine/src/manager/vine_current_transfers.c

```diff
@@ -11,27 +11,27 @@ See the file COPYING for details.
 
 struct vine_transfer_pair {
 	struct vine_worker_info *to;
-	char *source;
+	void *source;
```

Comment:
I used void * so it could still point to a char * for a static url file. eg. not a worker.

Reply:
We then determine the type by seeing if the worker_source field has been set in the vine_file object

Reply:
Too clever, I think it's better to clearly distinguish the use cases with proper types to avoid strange accidents, something like:
```
struct vine_worker_info *source_worker;
char *source_url;
```

Reply:
Or we can have two different functions as we seem to know whether the transfer comes from a worker or a generic URL.

Reply:
The operations on the different sources are separate with the exception of adding the transfer to the table in vine_manager_put_url where we needed to switch on the type of source. I am ok with having two separate fields it shouldn't be too cumbersome